src/main/java/de/tum/in/www1/artemis/web/rest/dto/CommitInfoDTO.java

```diff
@@ -0,0 +1,34 @@
+package de.tum.in.www1.artemis.web.rest.dto;
+
+import java.time.ZonedDateTime;
+
+import org.eclipse.jgit.revwalk.RevCommit;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+/**
+ * DTO for a commit info.
+ *
+ * @param hash      the hash of the commit
+ * @param message   the message of the commit
+ * @param timestamp the timestamp of the commit
+ * @param author    the author of the commit
+ */
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record CommitInfoDTO(String hash, String message, ZonedDateTime timestamp, String author) {
+
+    /**
+     * Creates a CommitInfoDTO from a RevCommit.
+     *
+     * @param commit the commit to create the DTO from
+     * @return the created DTO
+     */
+    public static CommitInfoDTO of(RevCommit commit) {
```

Comment:
Optional: I would prefer a constructor rather than calling a static method to generate an instance.

Reply:
In this case I kept the static method because otherwise I need to use a lot of duplicate calls to getters because the constructor call needs to be the first statement, so I cannot use local variables.